File: mobile-modules/truckdocnetwork/app/src/main/java/com/sanda/truckdoc/network/api/interceptors/AuthorizationInterceptor.java
package com.sanda.truckdoc.network.api.interceptors;

import android.util.Base64;

import com.sanda.truckdoc.network.api.UserKey;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by k.natallie on 04.11.2016.
 */

public class AuthorizationInterceptor implements Interceptor {
    @NotNull
    private UserKey userKey;

    public AuthorizationInterceptor(@NotNull UserKey userKey) {
        this.userKey = userKey;
    }

    private String encodeCredentialsForBasicAuthorization() {
        final String credentials = userKey.getLogin() + ":" + userKey.getSecret();
        return "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP);
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();

        String authorizationValue = encodeCredentialsForBasicAuthorization();
        Request authorizedRequest = originalRequest.newBuilder()
                .removeHeader("Authorization")
                .addHeader("Authorization", authorizationValue)
                .build();
        return chain.proceed(authorizedRequest);
    }
}

File: truckdoc-client-m2/application/src/main/java/com/sanda/truckdoc/client/data/api/interceptors/AuthorizationInterceptor.kt
package com.sanda.truckdoc.client.data.api.interceptors

import okhttp3.Interceptor
import okhttp3.Response
import javax.inject.Inject

class AuthorizationInterceptor @Inject constructor() : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request().newBuilder()
            .addHeader("Authorization", "Bearer ${getAuthToken()}")
            .build()
        return chain.proceed(request)
    }

    private fun getAuthToken(): String {
        // TODO: Implement token retrieval from secure storage
        return ""
    }
} 